libs/server/Objects/Hash/HashObject.cs

```diff
@@ -45,25 +45,23 @@ public unsafe partial class HashObject : GarnetObjectBase
         /// <summary>
         ///  HashObject Constructor
         /// </summary>
-        public HashObject()
+        public HashObject(long expiration = 0) : base(expiration, MemoryUtils.DictionaryOverhead)
         {
             hash = new Dictionary<byte[], byte[]>(new ByteArrayComparer());
-            this.Size = MemoryUtils.DictionaryOverhead;
         }
 
         /// <summary>
         /// Copy constructor
         /// </summary>
-        public HashObject(Dictionary<byte[], byte[]> hash, long size)
+        public HashObject(Dictionary<byte[], byte[]> hash, long expiration, long size) : base(expiration, size)
         {
             this.hash = hash;
-            this.Size = size;
         }
 
         /// <summary>
         /// Construct from binary serialized form
         /// </summary>
-        public HashObject(BinaryReader reader) : this()
+        public HashObject(BinaryReader reader, long expiration) : this(expiration)
```

Comment:
Why not pass in the reader to the base class so that it can handle reading expiration direclty? That way, the base class fields info can stay within the base class and not have to be exposed to the derived class.

Reply:
great idea - i have made this change to the PR.